visualization/include/pcl/visualization/impl/point_cloud_color_handlers.hpp

```diff
@@ -564,6 +485,232 @@ PointCloudColorHandlerLabelField<PointT>::getColor () const
   return scalars;
 }
 
+// ----
+// template specializations for PointCloud and PCLPointCloud2 to access certain
+// data from PointCloudColorHandlerGenericField without needing to know the
+// cloud type
+
+// Get point fields from cloud. Could not get it to work with existing
+// pcl::getFields
+template <typename CloudT> inline std::vector<pcl::PCLPointField>
+getFields(const CloudT& cloud)
```

Comment:
I would move these to the other getFields functions in io.h - so they are grouped there and not scattered around in various modules/files.

Reply:
But lets wait for @mvieth to answer as well, before moving it around.